crates/chia-consensus/src/spendbundle_conditions.rs

```diff
@@ -30,12 +31,22 @@ pub fn get_conditions_from_spendbundle(
     let mut ret = SpendBundleConditions::default();
     let mut state = ParseState::default();
 
-    for coin_spend in &spend_bundle.coin_spends {
-        let byte_cost = (coin_spend.puzzle_reveal.len() + coin_spend.solution.len()) as u64
-            * constants.cost_per_byte;
-
-        subtract_cost(a, &mut cost_left, byte_cost)?;
+    let spends_info: Vec<_> = spend_bundle
+        .coin_spends
+        .iter()
+        .map(|coin_spend| {
+            (
+                coin_spend.coin,
+                &coin_spend.puzzle_reveal,
+                &coin_spend.solution,
+            )
+        })
+        .collect();
```

Comment:
you don't need to `collect()` here. `solution_generator()` already takes an iterator.
However, since this is time-critical, I think it would be best to compute this length without also building the generator. Would that be unreasonably complex?

Reply:
> you don't need to collect() here. solution_generator() already takes an iterator.

Done.

> Would that be unreasonably complex?

I believe so.